### PR TITLE
`PlatformReleaser` plugin interface docs

### PR DESCRIPTION
**Pull request by paladin-devops**
This PR adds docs for the [`PlatformReleaser` interface of the plugin SDK](https://github.com/hashicorp/waypoint-plugin-sdk/blob/main/component/component.go#L119-L127)! 😄

### CODE REVIEW

**Review by evanphx — APPROVED** [review]

File: website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+page_title: 'PlatformReleaser'
+description: |-
+  How to implement the PlatformReleaser component for a Waypoint plugin
+---
+
+# PlatformReleaser
+
+https://pkg.go.dev/github.com/hashicorp/waypoint-plugin-sdk/component#PlatformReleaser
+
+The PlatformReleaser interface is an optional interface that a Platform component
```
Comment (catsby) [inline website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx:12]:
```suggestion
The `PlatformReleaser` interface is an optional interface that a Platform component
```

I think interface names like this should have the special font applied. I know we aren't very consistent here, but we should try to be IMO

File: website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+can implement to provide default Release functionality. This only takes effect
```
Comment (catsby) [inline website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx:13]:
```suggestion
can implement to provide default [`Release`](/docs/extending-waypoint/plugin-interfaces/release-manager) functionality. This only takes effect
```

I think this makes sense to add font and link, but this is just a suggestion
could also link this to

File: website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+if no release is configured. Therefore, users cannot set custom configurations
```
Comment (catsby) [inline website/content/docs/extending-waypoint/plugin-interfaces/platform-releaser.mdx:14]:
```suggestion
if no release is configured, therefore users cannot set custom configurations
```

IMO these read like 3 half-sentances in a row, combining these two keeps the chain of thought going

**Review by catsby — APPROVED** [review]
I added some minor suggestions but looks good without them, so do as you wish

**Pull request merged**